src/app/modules/products/components/product-detail/product-detail.component.ts

```diff
@@ -1,23 +1,60 @@
-import { Component, OnInit, Input } from "@angular/core";
+import { Component, Input } from "@angular/core";
+import { FormBuilder, Validators } from "@angular/forms";
+
 import { ImageService } from "@core/views/image.service";
 
 import { Product } from "@shared/models/classes/product.class";
+import { Weight } from "@shared/models/types/weight.type";
+import { Roast } from "@shared/models/types/roast.type";
+import { CustomOption } from "@shared/models/types/custom-option.interface";
 
 @Component({
     selector: "app-product-detail",
     templateUrl: "./product-detail.component.html",
     styleUrls: ["./product-detail.component.scss"]
 })
-export class ProductDetailComponent implements OnInit {
+export class ProductDetailComponent {
     @Input() product!: Product;
 
+    readonly weightOptions: Weight[] = ["250g", "1kg"];
+    readonly defaultWeight: Weight = this.weightOptions[0];
+    currentWeight: Weight = this.defaultWeight;
+
+    readonly roastOptions: Roast[] = ["Filter", "Espresso", "Both"];
+    readonly defaultRoast: Roast = this.roastOptions[0];
+    currentRoast: Roast = this.defaultRoast;
+
+    snipcartOptions: CustomOption[] = this.makeSnipcartOptions();
+
+    selectionForm = this.fb.group({
```

Comment:
the form is not used. We have two options:

1) we use form Controls for the radio buttons, then subscribe to onValueChanges and updage the value of snipcartOptions on form changes
2) we remove the form from the component and template and use the methods here. If you go this route, I would combine the onSelectWeight and onSelectRoast methods into one. You can pass a generic option variable. The currentWeight is only used in the options array. So it can all be done at once

Reply:
The same should be done in the product-card

Reply:
This all changed in the [Add-quantity-input-to-product PR](https://github.com/lpboucher/dak-coffee-wholesale/pull/74) where I think it got refactored to something much cleaner than what I have here. Should I just leave that to be merged in later, or would you like me to bring it in here as well?